src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx

```diff
@@ -0,0 +1,60 @@
+import type { UrlObject } from "url";
+
+import React from "react";
+import Link from "next/link";
+import styled from "styled-components";
+
+import {
+  OakPrimaryButton,
+  OakPrimaryButtonProps,
+} from "@/components/ui/OakPrimaryButton";
+import {
+  OakPrimaryInvertedButton,
+  OakPrimaryInvertedButtonProps,
+} from "@/components/ui/OakPrimaryInvertedButton";
+
+type Url = string | UrlObject;
+
+export type OakPrimaryNavItemProps = {
+  href: Url;
+  isCurrent?: boolean;
+  shallow?: boolean;
+  label: string;
```

Comment:
minor, but this should probably be `children` and the type `ReactNode` to fit into the components it's wrapping. As a consumer i'd expect the API to be the same as a regular button/link.